### ISSUE TITLE
Versioned AND concept classes for any entity

### ISSUE THREAD

**Issue by mih** [psychoinformatics-de/datalad-concepts#22]
#20 hints at a need for that. Any entity may have properties that may change with a new version (hence a new version is a new entity). At the same time, such a versioned entity may be a specialization/revision of a more general/eternal concept that does not change over time/versions.

For example, a `README` file in a dataset will always be a documentation component, and might always be meant use a particular language.

So similar to the idea of factoring out identical metadata from files in a directory into a dedicated `Directory` description (see #20), we might want to specifically distinguish versioned from unversioned representations of any recognized dataset component entity -- with the idea of ascribing persistent properties to them.

**Comment by mih** [comment]
Tried to convince myself that this is worth the effort. Looking at established efforts (e.g. #25), I can find similar conceptual distinctions (a `Dataset` being a conceptual entity, with a `Distribution` being its materialization). However, I found no example of using this distinction for the purpose of versioning.

Reasons I came up with as counter-arguments:

- any unversioned declaration of properties (purpose, permissions, whatever) runs the risk of invalidation with each (future) version
- the factual properties of an unversioned entity are few and more-or-less entirely restricted to the expression of version-relationships

So with that we might want to abandon any unversioned concepts, and limit the linkage of a Dataset(Version) to its "concept" to a `isVersionOf` property:

```yaml
id: gitsha:cdc0ceb30ae04265c5369186acf2ab2683a8ec96
is_version_of: dlds:eba5177f-2040-448f-b21d-1f0ae698d0f0
```
